kai/report.py

```diff
@@ -10,35 +10,61 @@
 
 
 class Report:
-    def __init__(self, path_to_report: str):
-        self.path_to_report = path_to_report
+    report = {}
+
+    def __init__(self, report_data: dict):
         self.workaround_counter_for_missing_ruleset_name = 0
-        ## We expect that self._read_report() will be the last statement
-        ## to invoke in this constructor
-        self.report = self._read_report()
+        self.report = self._reformat_report(report_data)
+
```

Comment:
This doesn't work intuitively. Consider this simplified case:

```python
class Report:
    report = {}

r1, r2 = Report(), Report()
r1.report['name'] = 'thing'
print(r2.report)  # Output: {'name': 'thing'}
```

Granted, self.report is set on line 17 so this is kinda mute, but just trying to avoid a gotcha. It might be better to just put a type hint like `report: dict`

Reply:
Good catch